File: FinalProject/app/src/main/java/com/example/finalproject/service/Triplex.java
package com.example.finalproject.service;

import android.util.Pair;

public class Triplex<T, T1, T2> {
    public T first;
    public T1 second;
    public T2 third;

    public Triplex() {
    }

    public Triplex(T first, T1 second, T2 third) {
        this.first = first;
        this.second = second;
        this.third = third;
    }

    public Pair<T, T1> getPair(){
        return new Pair<>(first, second);
    }

    @Override
    public String toString() {
        return "Triplex{" +
                "first=" + first +
                ", second=" + second +
                ", third=" + third +
                '}';
    }
}

File: FinalProject/app/src/main/java/com/example/finalproject/service/Triplex.kt
package com.example.finalproject.service

data class Triplex<T1, T2, T3>(var first:T1, var second:T2, var third:T3) {
    override fun toString(): String {
        return "Triplex{ frist=$first, second=$second, third=$third}"
    }

    fun getPair(): android.util.Pair<T1, T2> {
        return android.util.Pair<T1, T2>(first, second)
    }
}